tests/epyccel/modules/arrays.py

```diff
@@ -1753,3 +1745,32 @@ def multiple_np_linspace():
     y = np.linspace(0, 4, 128)
     z = np.linspace(0, 8, 128)
     return x[0] + y[1] + z[2] + linspace_index
+
+#==============================================================================
+# Iteration
+#==============================================================================
+
+def iterate_slice(i : int):
+    from numpy import arange
+    a = arange(15)
+    res = 0
+    for ai in a[:i]:
+        res += ai
+    return res
+
+@template('T', ['int[:]', 'int[:,:]', 'int[:,:,:]', 'int[:,:](order=F)', 'int[:,:,:](order=F)'])
+def unpack_array(arr : 'T'):
+    x, y, z = arr[:]
+    return x, y, z
+
+def unpack_array_of_known_size():
+    from numpy import array
+    arr = array([1,2,3], dtype='float64')
+    x, y, z = arr[:]
+    return x, y, z
+
+def unpack_array_2D_of_known_size():
+    from numpy import array
+    arr = array([[1,2,3], [4,5,6], [7,8,9]], dtype='float64')
+    x, y, z = arr[:]
+    return x[0], y[1], z[2]
```

Comment:
Why not writing `return x.sum(), y.sum(), z.sum()`? That uses all the values in the array.

Reply:
Fixed : [eb42d3c](https://github.com/pyccel/pyccel/pull/1793/commits/eb42d3c7cafd747be03a53b82733c280d5a00997)

I wasn't sure what to use to get scalars here (needed to not return a pointer to an out of scope object). I usually prefer to check values as these can't be misinterpreted whereas sum could theoretically return the correct value without summing the correct values. But I agree that it is better to use all the values